doc/plutus/explanations/plutus-tools-component-descriptions.rst

```diff
@@ -0,0 +1,241 @@
+
+   See also: 
+   
+   * :doc:`Property-based testing of Plutus contracts <../tutorials/contract-testing>` 
+   * :doc:`Testing Plutus Contracts with Contract Models <../tutorials/contract-models>` 
+
+3. Contract monad emulator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+---------------------------------------------------------------+
+| Contract monad emulator                                                      |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+
+`Contract monad emulator <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ is a library that provides an environment for emulating the blockchain. 
+The environment provides a way for writing traces for the contract which are sequences of actions by simulated wallets that use the contract. 
+The component is highly dependent on the Contract API (Contract monad). 
+
+4. Contract API (also known as Contract monad)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+---------------------------------------------------------------+
+| Contract API                                                                 |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+
+`Contract API <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ is a logical component within the Plutus Contract package, providing an effective system for describing smart contracts that interact with wallets, DApps and the blockchain. 
+It provides the Contract API interface for writing the off-chain part of a Plutus application that is to be interpreted by an emulator or by Plutus application backend (PAB). 
+
+5. Plutus chain index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+---------------------------------------------------------------+
+| Plutus chain index                                                           |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-chain-index-core``                       |
+|              | ``plutus-apps/plutus-chain-index``                            |
++--------------+---------------------------------------------------------------+
+
+   COMMENT: Ziyang suggests that we remove this one from the page. Does anyone have other opinions? 
+
+`Plutus chain index <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-chain-index-core>`_ 
+is an application for indexing data from the Cardano blockchain that is 
+used in the Contract Monad. The main design is for indexing information related to the 
```

Comment:
I think this sounds a little better: `The main design goal is to keep the size of the indexed information proportional to the Utxo set.`